### DISCUSSION TITLE
Shouldn't `getStaticPaths` have access to localization?

### DISCUSSION THREAD

**Discussion by mxmzb** [vercel/next.js#19499] [Ideas]
I'm wondering:

Shouldn't `getStaticPaths` have access to the `locale` parameter like `getStaticProps` have? My paths may in fact depend on the locale I'm in (e.g. I have an article written in `en` but not yet in `ru`), and there is no way to generate this difference in paths currently, at least from my understanding.

**Comment by ijjk** [comment]
Hi, `getStaticPaths` is only called once and is expected to return all paths and their locale variants. Since it is only called once we don't pass the specific `locale` value since there isn't one specific value being used at that time. If a `locale` isn't provided on a path returned from `getStaticPaths` the `defaultLocale` will be used. See more info on this in the [internationalized routing docs](https://nextjs.org/docs/advanced-features/i18n-routing#dynamic-getstaticprops-pages)
